src/components/organisms/AppRouter/AdminSubrouter.tsx

```diff
@@ -5,31 +5,100 @@ import Admin from "pages/Admin/Admin";
 import { RoomsForm } from "pages/Admin/Venue/Rooms/RoomsForm";
 import { VenueWizard } from "pages/Admin/Venue/VenueWizard";
 
-import Admin_v2 from "pages/Admin/Admin_v2";
-import VenueWizard_v2 from "pages/Admin/Venue/VenueWizard/VenueWizard";
+import AdminV2 from "pages/Admin/Admin_v2";
+import VenueWizardV2 from "pages/Admin/Venue/VenueWizard/VenueWizard";
 import { AdminAdvancedSettings } from "pages/AdminAdvancedSettings";
 
 import { AdminVenueView } from "components/organisms/AdminVenueView";
+import { Provided } from "components/organisms/AppRouter/Provided";
 
 export const AdminSubrouter: React.FC = () => {
   return (
     <Switch>
       {/* Admin V1 */}
-      <Route path="/admin/venue/rooms/:venueId" component={RoomsForm} />
-      <Route path="/admin/venue/creation" component={VenueWizard} />
-      <Route path="/admin/venue/edit/:venueId" component={VenueWizard} />
-      <Route path="/admin/:venueId" component={Admin} />
-      <Route path="/admin" component={Admin} />
+      <Route
+        path="/admin/venue/rooms/:venueId"
+        component={() => (
+          <Provided withWorldUsers withRelatedVenues>
+            <RoomsForm />
+          </Provided>
+        )}
+      />
+      <Route
+        path="/admin/venue/creation"
+        component={
+          // for some state related reasons, Provided fails this VenueWizard
+          // and using function here will cause the error
+          // so WithNavigationBar wraps the RelatedVenuesProvider for it
+          VenueWizard
+        }
+      />
```

Comment:
What were the issues/etc observed with this that made it 'fail'? Specifics of the errors, etc?

Reply:
Didn't note them at the time. Will add the provider to get exact error message.

Reply:
Steps to reproduce: 

1. replace the `component` prop with:
```
        component={() => (
          <Provided withWorldUsers withRelatedVenues>
            <VenueWizard />
          </Provided>
        )}
```
2. head to URL ending in `/admin/venue/creation?page=1`
3. pick a space type, and click `Create your space` button

Error in browser console:
```
Warning: Cannot update a component (`WithNavigationBar`) while rendering a different component (`DetailsForm`). To locate the bad setState() call inside `DetailsForm`, follow the stack trace as described in https://fb.me/setstate-in-render
    in DetailsForm (at VenueWizard.tsx:133)
    in div (at WithNavigationBar.tsx:46)
    in WithNavigationBar (at VenueWizard.tsx:143)
    in VenueWizardCreate (at VenueWizard.tsx:61)
    in VenueWizard (at AdminSubrouter.tsx:31)
    in RelatedVenuesProvider (at Provided.tsx:22)
    in WorldUsersProvider (at Provided.tsx:21)
    in Provided (at AdminSubrouter.tsx:30)
    in component (created by Context.Consumer)
    in Route (at AdminSubrouter.tsx:27)
    in Switch (at AdminSubrouter.tsx:17)
    in AdminSubrouter (created by Context.Consumer)
    in Route (at AppRouter.tsx:76)
    in Switch (at AppRouter.tsx:73)
    in Suspense (at AppRouter.tsx:72)
    in Router (created by BrowserRouter)
    in BrowserRouter (at AppRouter.tsx:71)
    in AppRouter (at src/index.tsx:283)
```

Reply:
Same or similar error will show up with or without `RelatedVenuesProvider` in `WithNavigationBar`

Reply:
And that error doesn't show up when you don't wrap it like this? That sounds like an error within that component itself rather than an issue with how we're doing things here.

I ran into similar at some point during my refactoring of the 'account onboarding' flow. Can't remember exactly, but I think the places where it happened it was due to code trying to change navigation/etc in the 'main body' of the component (which is invalid React), rather than using a `useEffect` or similar to wrap it.

Reply:
Replicated following instructions in https://github.com/sparkletown/sparkle/pull/1769#discussion_r667280484

---

> And that error doesn't show up when you don't wrap it like this? 

Seems the error doesn't show up given the currently committed syntax (which is strange and a little confusing to me.. but anyway..

Looking further into this now.

Reply:
Ok, pretty sure I have found it:

`src/pages/Admin/Venue/DetailsForm.tsx:272` calls the `previous()` function in the body of the component.

This is defined at `src/pages/Admin/Venue/VenueWizard.tsx:130` and uses `history.push` to navigate to a new page.

![image](https://user-images.githubusercontent.com/753891/125152373-7e45ab00-e18f-11eb-8a2e-3a42998435b6.png)


I can wrap it in `useEffect` properly and the error goes away, but need to dig into the expect logic of this section of the admin panel more, because once I 'fix' that, the rest of the flow seems broken since I always get sent back to that first menu page because `state.templatePage` doesn't ever seem to be defined for me.

Reply:
> once I 'fix' that, the rest of the flow seems broken since I always get sent back to that first menu page because `state.templatePage` doesn't ever seem to be defined for me.

Ok, got this figured out, and it's basically a case of the exact problem I called out in https://github.com/sparkletown/sparkle/pull/1769#discussion_r667270203 where using the `component={() => (` syntax  would cause issues because it will always unmount/re-mount the component.

So in this case, the `state` is empty because it's been unmounted and remounted.

But by properly changing this `Route` to use the 'nested' style + wrapping the navigation in a `useEffect` within the component itself, it seems to fix it.

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1769/commits/cc0930efb0a06d0a1a6ba22e43128973f681b427 + https://github.com/sparkletown/sparkle/pull/1769/commits/4fd34816ca35db8046e256cda95b6e842bce3684